enumerable.rb

```diff
@@ -0,0 +1,188 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    (0..length - 1).each do |i|
+      yield(self[i])
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    (0..length - 1).each do |i|
+      yield(self[i], i)
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    new_arr = []
+    (0..length - 1).each do |i|
+      new_arr.push(self[i]) if yield(self[i])
+    end
+    new_arr
+  end
+
+  def my_all?(*arg)
+    if block_given?
+      my_each do |i|
+        return false unless yield(i)
+      end
+    elsif !arg.empty?
+      if arg[0].is_a? Numeric
+        my_each do |i|
+          return false if i != arg[0]
+        end
+      elsif arg[0].class == Regexp
+        my_each do |i|
+          return false if i !~ arg[0]
+        end
+      else
```

Comment:
**[OPTIONAL]**
You should create a private method to do this check since you do in 2 of your methods. Something like `check_pattern`.